packages/validator/src/validate.ts

```diff
@@ -0,0 +1,49 @@
+import fs from 'fs';
+import * as logger from './utils/logger';
+import { Result } from './types';
+import { validateSpecs } from './validator';
+
+/**
+ * Validates specification from provided file according to template file
+ * @param specsPath - specification to validate, root must be an object (not an array)
+ * @param templatePath - validator specification structure
+ * @returns array of error and warning messages
+ */
+export function validate(specsPath: string | undefined, templatePath: string | undefined): Result {
+  if (!specsPath || !templatePath) {
+    return { valid: false, messages: [logger.error('Specification and template file must be provided')] };
+  }
+
+  let template, specs;
+
+  try {
+    template = fs.readFileSync(templatePath);
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`Unable to read file ${templatePath}`)] };
+  }
+
+  try {
+    specs = fs.readFileSync(specsPath);
+  } catch (e) {
+    return { valid: false, messages: [logger.error(`Unable to read file ${specsPath}`)] };
+  }
+
+  try {
+    const nonRedundant = {};
+    const parsedTemplate = JSON.parse(template);
+    const parsedSpecs = JSON.parse(specs);
+
+    return validateSpecs(parsedSpecs, parsedTemplate, '', nonRedundant, {
+      specs: parsedSpecs,
+      nonRedundantParams: nonRedundant,
+    });
+  } catch (e) {
+    return { valid: false, messages: [{ level: 'error', message: `${e.name}: ${e.message}` }] };
+  }
+}
+
+if (process.env.npm_config_specs || process.env.npm_config_template) {
+  console.log(validate(process.env.npm_config_specs, process.env.npm_config_template));
+} else {
+  console.log(validate(process.argv[2], process.argv[3]));
```

Comment:
Does process.argv work when providing arguments in the form of `--variable=value`. I thought this would mean they come through as just that, so `process.argv[2]` would be `--specs=config.json` when we want just `config.json`.

It might be worth using a very basic [yargs](https://github.com/yargs/yargs) setup. Burak is also using yargs the `aws-deployment` branch, although that usage is much more complex

Reply:
If arguments are passed as `--variable="value"` first part of condition is executed, otherwise it will try to retrieve regular whitespace separated arguments and execute with them.
yargs seems like powerful tool, isnt it a bit overkill for this though? Imho this is a simple and elegant solution for this problem, allowing to specify arguments in 2 different ways without any robust 3rd party tools.

Reply:
Yeah I'm happy with this then. No need for yargs